src/openloco/utility/numeric.cpp

```diff
@@ -0,0 +1,33 @@
+#include "numeric.hpp"
+
+#ifdef _WIN32
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#endif
+
+#include <cstddef>
+#include <cstdint>
+#include <limits.h>
+#include <type_traits>
+
+namespace openloco::utility
+{
+    int32_t bitscanforward(int32_t source)
```

Comment:
wouldn't it make more sense to have an `uint32_t` as argument? Also, I find 'bitscanforward' a bit unclear. Something like `findFirstBitSet` sounds more logical to me.

Reply:
Oh, I completely agree there. This is the same function used in OpenRCT2. Wonder why they use an int32_t in the signature, only to cast it to an uint32_t later.

As for the name, I agree it's not the best one, but as it's directly intended to replace a `bsf` instruction, I think it would be better to keep the name for discoverability. I'll happily add a few lines explaining the function, though.

Reply:
👍